crates/ironrdp-async/src/framed.rs

```diff
@@ -236,39 +230,44 @@ pub async fn single_sequence_step<S>(
     framed: &mut Framed<S>,
     sequence: &mut dyn Sequence,
     buf: &mut WriteBuf,
-    unmatched: Option<&mut Vec<Bytes>>,
 ) -> ConnectorResult<()>
 where
     S: FramedWrite + FramedRead,
 {
     buf.clear();
-    let written = single_sequence_step_read(framed, sequence, buf, unmatched).await?;
+    let written = single_sequence_step_read(framed, sequence, buf).await?;
     single_sequence_step_write(framed, buf, written).await
 }
 
 pub async fn single_sequence_step_read<S>(
     framed: &mut Framed<S>,
     sequence: &mut dyn Sequence,
     buf: &mut WriteBuf,
-    unmatched: Option<&mut Vec<Bytes>>,
 ) -> ConnectorResult<Written>
 where
     S: FramedRead,
 {
     buf.clear();
 
     if let Some(next_pdu_hint) = sequence.next_pdu_hint() {
+        let ignore_pdu_error = sequence.ignore_pdu_error();
         debug!(
             connector.state = sequence.state().name(),
             hint = ?next_pdu_hint,
+            ?ignore_pdu_error,
             "Wait for PDU"
         );
 
         let pdu = framed
-            .read_by_hint(next_pdu_hint, unmatched)
+            .read_by_hint(next_pdu_hint)
             .await
-            .map_err(|e| ironrdp_connector::custom_err!("read frame by hint", e))?;
+            .map_err(|e| ironrdp_connector::custom_err!("read frame by hint", e));
 
+        if ignore_pdu_error && pdu.is_err() {
+            return Ok(Written::Nothing);
+        }
+
+        let pdu = pdu?;
```

Comment:
question: I’m not sure we need the new `ignore_pdu_error` thing?
`read_by_hint` is not decoding PDUs, and unexpected frames (based on the "next PDU hint"), are already ignored.
The `DecodeError` itself is handled inside the acceptor sequence. Am I missing something?

Reply:
It would raise an error, either during read_by_hint or later DecodeError. This aborts accept_finalize().

Reply:
Do you know why? What kind of payload are we receiving here which may cause accept_finalize() to fail?

Reply:
That can be pretty much any PDU inflight during reactivation-reactivation. Server sends DemandActive, but anything can be in flight.

Reply:
We still need to be framing packets properly, otherwise we wouldn’t know where to start reading the next relevant packet. And for that matter, all active sharing packets are [either X224 (SlowPath) or FastPath](https://github.com/Devolutions/IronRDP/blob/dd221bf22401c4635798ec012724cba7e6d503b2/crates/ironrdp-pdu/src/lib.rs#L93-L118).
The ConfirmActive PDU which is sent by the client after stopping sending any other active share PDU is one of such packets (X224?).

Reply:
It can/does fail during single_sequence_step(). However it's true that PduHint::find_size() shouldn't fail since all incoming packets should be either x224 or FP. Thus we could probably remove the ignore_pdu_error(). I think I have seen it happen though, not sure how/when. If you feel strongly about it, I'll drop it.

Reply:
`single_sequence_step()` has multiple point of failure other than `read_by_hint()`, including `sequence.step()`.
I suspect the real culprit of the failure was `sequence.step()`, not `read_by_hint()`.
Now it is handling unexpected PDUs until a safe point.
Yes, I would rather drop the new API surface to keep the code as simple as possible.
I do agree that defensive patterns are useful when dealing with network, but I don’t think it’s worth being that defensive here.
Thank you for all the efforts!